src/cmd/sign.js

```diff
@@ -107,6 +117,27 @@ export default function sign(
         downloadDir: artifactsDir,
       });
 
+      if (updateLink) {
+        let extensionID;
+        if (signingResult.id) {
+          extensionID = signingResult.id;
+        } else {
+          extensionID = id;
+        }
+        // flow thinks that oldUpdateManifestData can be potentially undefined
+        // this is wrong however, the if-case guarding both the `oldUpdateManifestData = `
+        // statement and this function call are both checking
+        // the exact same thing, which doesn't change
+        // anywhere in the code. Basically, `if (A) then define B` and later on `if (A) then use B`
+        // which flow doesn't seem to understand
+        let oldUpdateManifestDataMakeFlowHappy =
+          oldUpdateManifestData == null ? {} : oldUpdateManifestData;
+        await createNewUpdateManifest(
+          extensionID, signingResult.downloadedFiles[0], artifactsDir,
```

Comment:
ok, I looked more into the case where multiple files could exist in `signingResult.downloadedFiles` and this is just for legacy (non-webextension) add-ons. So, yeah, it should never happen. However, I'd like to see something that checks the length of this array and throws an error if it's longer than one. The error should just explain how such a situation is not supported.